docs/user-manual/en/security.md

```diff
@@ -1221,6 +1221,11 @@ management performed on the live broker will be reflected on the backup upon
 failover. Typically LDAP would be employed for this kind of use-case, but not
 everyone wants or is able to administer an independent LDAP server.
 
+> **Note:**
+>
+> `ActiveMQBasicSecurityManager` works only with standalone broker 
+> configuration or with live/backup clusters. It won't work with multiple lives.
+
```

Comment:
You mean like JAAS _PropertiesLogin_ manager?

Reply:
To be clear, the `PropertiesLoginModule` is a login module used with the JAAS security manager. There is no such thing as a "JAAS PropertiesLogin manager."

Regarding the basic security manager your documentation update says, "It won't work with multiple lives." My point is that the basic security manager works _exactly_ like the JAAS security manager in regard to multiple lives. _Neither one_ will propagate credentials around the cluster. Furthermore, you say, "...works only with standalone broker configuration..." Again, this is unclear because you _can_ use the basic security manager in a cluster just like you can with the JAAS security manager. The **main use** (if not the only use) where the basic security manager makes sense is with a single live/backup pair.

The documentation already explains this fact saying:

> All user & role data is stored in the bindings journal (or bindings table if using JDBC). The advantage here is that in a live/backup use-case any user management performed on the live broker will be reflected on the backup upon failover.

Although perhaps it could be stated more strongly.

Reply:
I ment JAAS security manager _when using PropertiesLogin module_. All of the security managers work in a cluster per se obviously, but it doesn't mean that the result for the end user/Artemis administrator is the same. At least during my tests credential propagation works three different ways:
1. ActiveMQBasicSecurityManager propagates credentials in live/backup pair but not on other lives.
2. JAAS security manager with PropertiesLogin module doesn't propagate credentials at all because (at least by default) properties files are stored on one node.
3. JAAS security manager with LDAP module can be used with multiple lives and backups.

All of these can be unclear for newcomers since they do not know how exactly Artemis internals work. That's why I think it is a good idea to mention this somewhere.

I will try to reword my patch a bit.

Reply:
Its not clustered it is a local file, but if youre using a mechanism to auto configure your brokers like ansible or use rsync then you can easily keep the files in sync. Its not different than needing to keep all other config in sync in a clustered artemis setup.

Reply:
> Its not clustered it is a local file, but if youre using a mechanism to auto configure your brokers like ansible or use rsync then you can easily keep the files in sync. Its not different than needing to keep all other config in sync in a clustered artemis setup.

I know that there are mechanisms and workarounds, how one could synchronize configuration files, but we are talking about standard Artemis documentation here. I would rather have documentation explain how various configuration options work without involving workarounds.

Reply:
They aren't workarounds you litterally have to keep config of all your brokers updated anyhow...theres no magical wand. The security part is no different / no exception. As such one would expect and thus why such configuration management tools exist such as ansible, puppet, chef, etc etc.

Reply:
If anything simply should be advocating that anything above a single non ha setup, user should be looking to adopt a configuration management tool to ensure configuration across clustered hosts are managed consistently. This said I would hope that this is a given for any ops or sre.

Reply:
> I ment JAAS security manager _when using PropertiesLogin module_.

That's fair enough, but that point isn't clearly made in your update.

> I will try to reword my patch a bit.

Sounds good. Thanks!

Reply:
Sorry it took some time to get back to this.

How about now?

Reply:
I still think it's unclear. The way it's currently worded folks are going to think they can't use it at all with multiple lives when they, in fact, can albeit with the same limitations as with JAAS security manager with the PropertiesLoginModule (i.e. the default configuration). Perhaps you should just add a note in the documentation for the LDAPLoginModule that it's the only suitable option for sharing a security repository among multiple live brokers.

Reply:
And now?

Reply:
There's no substantive difference in your update vs. the previous version. This is the problematic bit:

> ActiveMQBasicSecurityManager` is not suitable for clusters with multiple lives.

This is calling out a limitation of the basic security manager that is also true of the default configuration (i.e. the JAAS security manager with the PropertiesLoginModule). However, since this limitation is not explicitly called out for the default configuration it makes it appear that this limitation is unique to the basic security manager.

I think something like this would be much more clear. Feel free to include this text in your commit if you like.

> All user & role data is stored in the broker's bindings journal (or bindings table if using JDBC). The advantage here is that in a live/backup use-case any user management performed on the live broker will be reflected on the backup upon failover. 
>
> Typically LDAP would be employed for this kind of use-case, but not everyone wants or is able to administer an independent LDAP server. One significant benefit of LDAP is that user data can be shared between multiple live brokers (e.g. in a cluster). However, this is not possible with the basic security manager or, in fact, any other configuration potentially available out of the box. Nevertheless, if you just want to share user data between a single live/backup pair then the basic security manager may be a good fit for you.

Reply:
For me this is the same meaning but with different wording :) Anyway, let's not break spears here. I have included your (slightly modified) text in that part of the change.